### PR TITLE
Move plugins

### PR DESCRIPTION
**Pull request by sagikazarmark**
Although there will be no version two, I added it so that in any case, we can continue using this package if necessary (like a metapackage installing all plugins, discussed elsewhere).

Also, please let me know what you think about the compatibility layer:
- Old plugin interface extends the new one, no API change
- Old plugin client wraps the new one

### CODE REVIEW

File: src/PluginClient.php [hunk 1]
```diff
@@ -2,43 +2,24 @@
...
-use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * The client managing plugins and providing a decorator around HTTP Clients.
  *
  * @author Joel Wurtz <joel.wurtz@gmail.com>
+ *
+ * @deprecated since since version 1.1, and will be removed in 2.0. Use {@link \Http\Client\Common\PluginClient} instead.
  */
 final class PluginClient implements HttpClient, HttpAsyncClient
```
Comment (dbu) [inline src/PluginClient.php:19]: could we simply extend the base plugin client and add nothing? as the plugin interface extends the base plugin interface, everything should "just work". or is the base client final?

**Comment by dbu** [comment]
looks good. and we actually should release that as 1.1 at some point. otherwise people that use stable deps will never see the deprecations in their code.

we could also add the trigger_error code like symfony does, to see deprecation errors when enabled. https://knpuniversity.com/blog/upgrading-symfony-2.7

File: src/PluginClient.php (hunk 1 above)
Comment (sagikazarmark) [inline src/PluginClient.php:19]: No we can't:
- The base client is final
- We would still have to do some hacks to throw the correct exception (in which case the exception is also final, so extension is not an option there)

**Comment by sagikazarmark** [comment]
> we could also add the trigger_error code like symfony does

Ah, wanted to, but forgot it.

**Comment by sagikazarmark** [comment]
Merge and tag?

**Comment by dbu** [comment]
:+1: i let you tag if all is ready.

**Comment by sagikazarmark** [comment]
Hm, what else to do?

Only the separate packages are not tagged, everything else is.

**Comment by dbu** [comment]
i think we should tag those (if they are ready to be tagged) before we tag this. when people discover they should move, the move target should be released and ready :-)

**Pull request merged**